Go ahead, Code!

zvenon opened the pull request:
So hab ich mir das mal vorgestellt fürs Erste 👌 
Könnt ja mal schreiben, was ihr davon haltet.
Code heut Abend noch weiter. Morgen kommt ein Update. 👋 

close #31 
close #33 
#32 habe ich jetzt mal so bearbeitet. Muss nochmal nachlesen zu Kotlin wegen der Liste oder vielleicht hat einer von euch einen Einfall.

zvenon committed:
Go ahead, Code!

zvenon commented:
Kommentare im Code sind in English. Sorry vorab, meins ist nicht soo gut, aber wollte im Code eine Sprache beibehalten. greets

zvenon commented on src/Verkehr/Network.kt:
```diff
@@ -0,0 +1,17 @@
+    /**
+     * initialize of the variables. start(?) value is zero
+     * variable cars represent the list/ number of cars
+     * constructor(from java).. try to explain in english: for initialisation of values which are relevant for our object (here. cars and capacity)
```
Kenne von JavaScript die constructo r Methode. Ist gerade in Arbeit. Kommt morgen. Hatte es schon kommentiert, nicht verwirren lassen ;)

zvenon committed:
Autos zählen/ Objekte einbinden

zvenon commented on src/Verkehr/Car.kt:
```diff
@@ -15,4 +15,31 @@ class Car {
+    fun randomizeWantsToDrive() {
+        wantsToDrive = Random().nextBoolean()
+    }
+
```
Das gehört quasi zu #34. Hab hier "random" festgelegt. Ist eine Methode, die eine Pseudozufallszahl zwischen 0 und 1 zurückgibt. Stellt im Prinzip die Simulation her, dass jedes Auto weiß, ob es fahren möchte oder nicht.

zvenon commented:
close #36 
close #34

zvenon commented:
close #34 
close #36

nipafx commented:
Die Issues, die du referenzierst sind alle schon geschlossen. Normalerweise wird ein Issue angelegt, dann erstellt man einen Pull Request, der eine Lösung vorschlägt und wenn sich alle einig sind, dass die Lösung gut ist, wird germergt und das Issue geschlossen. Ihr scheint einen anderen Vorgang zu haben, aber mir ist nicht klar, welcher.

nipafx commented:
Der Code ist ok. Wenn ich wüsste, was genau ihr in diesem PR erreichen wolltet, könnte ich sagen, ob ihr schon alles habt.

zvenon commented:
Oh man. Ja, das war dumm 🙈 . Hab die Issues wieder geöffnet. Jetzt hab ichs aber geschnallt :D

zvenon committed:
Simulation/Scenario eingebunden. Announcement dabei.

zvenon commented:
Der PR bezieht sich auf den Code zur Verkehrssimulation. Hab den step by step aktualisiert um Verbesserungsvorschläge/Änderungen am bisherigen Code entgegen zu nehmen.
Habe mir die Aufgabenstellung in OLAT durchgelesen, Konzept überlegt und anhand dessen die Issues erstellt. 
Die letzte Aktualisierung bezieht sich auf die Issues #37  #38  #39  #40 
Damit müsste der Code zur Simulation fertig sein(?!)
Hoffe ich hab die Aufgabenstellung oder die Vorgehensweise zum codieren oder lösen der Aufgabe nicht falsch verstanden :/

nipafx commented on src/Verkehr/Network.kt:
```diff
@@ -0,0 +1,70 @@
+        if(carsOnTheRoad <= capacity) {
+            //here starts the loop for delaying or driving
+            for(car in cars) {
+                if(car.wantsToDrive == true) {
```
Das hier meinte ich mit "the car announces whether it wants to drive". Die Methode `announceWantsToDrive` in `Car` kann weg.

nipafx commented on src/Verkehr/Network.kt:
```diff
@@ -0,0 +1,70 @@
+            }
+        } else {
+            for(car in cars){
+                car.delayed = true
```
Wenn du `drive` per Methode abwickelst, wäre es schöner, wenn du das für "verspätet werden" auch machst.

nipafx commented on src/Verkehr/Network.kt:
```diff
@@ -0,0 +1,70 @@
+     */
+
+    var capacity = 0
+    var carsOnTheRoad = 0
```
Das Feld macht den Code etwas schwerer zu verstehen. Einfacher wäre es, wenn es wegkommt, `countCarsOnTheRoad` seine das Ergebnis seiner Berechnung zurück gibt und `checkCapacityAndCompare` das Ergebnis einer lokalen Variable zuweist.

nipafx commented on src/Verkehr/Network.kt:
```diff
@@ -0,0 +1,70 @@
+     *
+     */
+
+    fun checkCapacityAndCompare() {
```
Sowas wie `simulate` oder `simulateTraffic` oder so wäre ein besserer Name, meinst du nicht?

nipafx commented on src/Verkehr/Main.kt:
```diff
@@ -0,0 +1,53 @@
+ * output: true/false for delayed or driving + car id
+ * delayed or driving is in the Network.kt (there: the loop below)
+ * */
+fun scenario() {
```
Die Datei hier kann aktuell vier Dinge:

* Autos und ein Netzwerk erstellen
* für jedes Auto ausgeben ob es fahren will
* drei Simulationsschritte berechnen
* für jedes Auto ausgeben, ob es gefahren ist

Das ist prinzipiell in Ordnung, aber sehr verstreut und aktuell von nirgendwo aufzurufen. Könntest du diese Methode hier so erweitern, dass sie alles vier macht (ein Simulationsschritt reicht aber) und von einer neu zu schreibenden `main` Methode aufgerufen wird?

nipafx commented on src/Verkehr/Network.kt:
```diff
@@ -0,0 +1,70 @@
+                }
+            }
+        } else {
+            for(car in cars){
```
Nur Autos, die fahren wollen, können verspätet sein.

nipafx reviewed (changes requested):
Alles gut. Einige kleine Ausbesserungen in `Car` und `Network`. In `Main` muss ein bisschen mehr getan werden.

yddaPyo committed:
Add files via upload

Die Dateien gingen etwas an der Aufgabe vorbei, jedes Auto sollte nur sagen, ob es fahren will und dann ob die Straße überlastet ist. Ich habe außerdem mit dem Parser Teil angefangen erstelle dazu jetzt ein Issue.

nipafx commented:
Meld dich, wenn du alle meine Kommentare erledigt hast.

yddaPyo committed:
Add files via upload

yddaPyo commented:
Habe nicht in github geschaut und habe versucht csv einzubauen, werde jetzt deine Änderungen vornehmen

yddaPyo committed:
Implementierung von CSV

Implementierung von csv Auslesung sowie die von Nicolai geforderten Änderungen

yddaPyo committed:
Einbinden der read und writer Funktionen

Habe vergessen die Funktionen zu verwenden, ist nun implementiert

yddaPyo committed:
Delete Network.kt

yddaPyo committed:
Delete Car.kt

yddaPyo committed:
Delete AccessToFile.kt

yddaPyo committed:
Delete Main.kt

yddaPyo committed:
Implementierung der Scenario Funktion

zvenon committed:
Application README hinzugefügt. Test funktioniert.

nipafx commented:
Statt den Pull Request um unzusammenhängende Änderungen (b42ae75) zu erweitern, wäre es sinnvoller:

* hier meine Änderungswünsche anzugehen
* andere Änderungen in anderen PRs durchzuführen

archdtd111 commented:
Passt der Rest?

nipafx commented:
Ne, meine Kommentare sind sind ja noch nicht alle adressiert. Besonders [der hier](https://github.com/yddaPyo/Info-Klausur/pull/41#discussion_r132533444) ist wichtig.

archdtd111 committed:
Ausgabe 2.0-SNAPSHOT bitte ignorieren
